src/vs/workbench/contrib/files/browser/fileCommands.ts

```diff
@@ -638,17 +638,20 @@ KeybindingsRegistry.registerCommandAndKeybindingRule({
 						},
 						'languageId': {
 							'type': 'string'
+						},
+						'path': {
+							'type': 'string'
 						}
 					}
 				}
 			}
 		]
 	},
-	handler: async (accessor, args?: { languageId?: string; viewType?: string }) => {
+	handler: async (accessor, args?: { languageId?: string; viewType?: string; path?: string }) => {
 		const editorService = accessor.get(IEditorService);
 
 		await editorService.openEditor({
-			resource: undefined,
+			resource: args?.path ? URI.from({ scheme: Schemas.untitled, path: `Untitled-${args.path}` }) : undefined,
```

Comment:
This might break untitled assumptions: either it should be an untitled with associated path on disk or not set a resource at all, why is this done?

Reply:
@lramos15

Reply:
I see for what it was added for. I am a bit worried that this could break the following regex:

https://github.com/microsoft/vscode/blob/e0a65a97d4f349cf11a7cae804a5553ccb412528/src/vs/workbench/services/untitled/common/untitledTextEditorHandler.ts#L88

So far, we had the assumption that `Untitled-(number)` is for untitled without associated resource and anything else has an associated resource. Untitled with associated resources typically do not ask where to save to because of the path.

Reply:
Hmm is there a way we could make this work? This is a very nice experience so far